apiserver/facades/client/client/filtering.go

```diff
@@ -282,8 +282,12 @@ func buildUnitMatcherShims(u *state.Unit, patterns []string) []closurePredicate
 
 func matchPortRanges(patterns []string, portRanges ...network.PortRange) (bool, bool, error) {
 	for _, p := range portRanges {
+		splitPortStr := strings.Split(p.String(), "/")
+		pNum := splitPortStr[0]
 		for _, patt := range patterns {
-			if strings.HasPrefix(p.String(), patt) {
+			splitPattStr := strings.Split(patt, "/")
+			pattNum := splitPattStr[0]
+			if strings.EqualFold(pNum, pattNum) && strings.HasPrefix(p.String(), patt) {
```

Comment:
why do we need `EqualFold`? These are just numbers (right?), so equality should suffice. In which case we don't need the `HasPrefix` check either. i.e
```go
if pNum == pattNum {
```

Reply:
Thx, fixed